connect/mirror/src/main/java/org/apache/kafka/connect/mirror/MirrorCheckpointTask.java

```diff
@@ -190,4 +227,102 @@ public void commitRecord(SourceRecord record) {
             Checkpoint.unwrapGroup(record.sourcePartition()),
             System.currentTimeMillis() - record.timestamp());
     }
+
+    private void refreshIdleConsumerGroupOffset() {
+        Map<String, KafkaFuture<ConsumerGroupDescription>> consumerGroupsDesc = targetAdminClient
+            .describeConsumerGroups(consumerGroups).describedGroups();
+
+        for (String group : consumerGroups) {
+            try {
+                if (consumerGroupsDesc.get(group) == null) {
+                    // if consumerGroupsDesc does not contain this group, it should be the new consumer
+                    // group created at source cluster and its offsets should be sync-ed to target
+                    newConsumerGroup.add(group);
+                    continue;
+                }
+                ConsumerGroupDescription consumerGroupDesc = consumerGroupsDesc.get(group).get();
+                // consider to sync offset to the target cluster, only if the consumer group is 'idle'
+                if (consumerGroupDesc.state().equals(ConsumerGroupState.EMPTY)) {
```

Comment:
Should we also include group state `DEAD`? New groups in the source cluster appear as dead in the target cluster.

Reply:
include the `DEAD` state for the testing, check out the latest.

If there is no real consumer consuming from the target cluster, the state could be DEAD. I am interesting to see what does the status change to, after the real consumer consuming from the target cluster

Reply:
From my tests it doesn't seam to work. The CG doesn't show up in the target cluster when listing with `kafka-consumer-groups.sh`. Also, when I start a consumer it resets the offset to what is configured in the consumer (latest in my case).

Reply:
ok, I will debug it when I get some dedicated timeframe